### ISSUE TITLE
What are the biggest differences between this and my `doasedit` I packaged a while ago?

### ISSUE THREAD

**Issue by magnus-ISU** [monesonn/doasedit#1]
Repo link: https://gitlab.com/magnustesshu/doasedit

As far as I can tell just from reading the code, the main one is that this creates the temporary file with some extra random letters. I initially thought this was stupid, however I suppose if you have multiple users using `doasedit` at the same time it will offer some convenience. I would prefer if the extension was predictable though, such as the user's name. Let me know if there are any good arguments against this, though.

You also do `set -eu`, what does this do?

There may be other, subtle differences.

I would prefer to unify and create a single project; I know that before I packaged `doasedit` people had been making such scripts for years, and yet the most advanced script didn't have even some basic protections which I ensured. I also noticed I have a couple forks which I need to check out since they didn't create pull requests.

As it stands, it is very misleading I think to have a `doasedit` and a `doasedit-git` which are completely different projects, both on the AUR.

**Comment by nyxnor** [comment]
Hi Magnus, I am not the repo owner but Ive been digging doasedit scripts.

Your code has bash extensions and some commands are only applicable to your operating system and shell, `stat` options vary on different unix systems. Bashisms are not good for portability reasons.
You also hardcoded the `doas` path to `/sbin` which may not apply to everybody but I understand it is a restricted folder in your system, but on openbsd it is on /usr/bin/doas.
You also hardcoded on the shebang the path to bash, his code searches the environment for that path of `sh`.

The monesonn code is almost posix compliant (except for /dev/urandom, mktemp, xxd, readlink), meaning it runs on any posix shell, `sh` (not a shell), `bash`, `ksh`, `dash`, `ash`, `posh` and others.
To learn the `set` command read its [manual](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_25).

Besides that, I don't notice much difference.

**Comment by magnus-ISU** [comment]
Alright, I've removed the hardcoded `/sbin` path, you're correct that there was no reason for that except to make it less portable. `/sbin` is just a symlink to other `/usr/bin` on arch, but I assumed that as a security-minded program doas would be there on all systems. Dumb assumption.

I'm also aware that the `stat` code is hacky as hell and would very much like to remove it, *assuming I can get the same functionality in a portable way*. Does this repository or another doasedit do that?
As for removing bashisms, I think I initially started trying to write it in portable shell and couldn't figure out how to do something (probably the hacky stat stuff) and decided to just use bash.

Read the manual of `set`, sorry for not RTFMing before asking.

I guess that doing `env` in the shebang is probably a good idea. I wrote my script expressly for arch, so just used some arch things. But if I can fix the stat code I will change that as well, as there is no harm in doing it.
